### PR TITLE
set up gitpod builds for every PR

### PR DESCRIPTION
**Pull request by keshavsethi**
Here I am just testing gitpod with some changes
Thanks!

fixes #8122

### CODE REVIEW

**Comment by jywarren** [comment]
Just noting i added some database configs to https://github.com/publiclab/plots2/pull/8122 too, so let's see...

File: config/dabase.yml.gitpod [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Warning: The contents of the database defined as "test" will be erased and
```
Comment (jywarren) [inline config/dabase.yml.gitpod:1]: ooh! yeah..... good call!

**Comment by jywarren** [comment]
also try all of these! https://github.com/publiclab/plots2/pull/8122/files#diff-4e0c28ccf9ba5766935eb3f7e564693dR18-R26

**Comment by keshavsethi** [comment]
@jywarren I think it will setup db now Please review
Thanks!!

**Comment by jywarren** [comment]
wow! Hope so! 🎉 thank you! My build has taken >24m so far, in https://github.com/publiclab/plots2/pull/8122

but once prebuilt, hopefully it'll go faster.

**Comment by jywarren** [comment]
OK, i had to, in the console, run:

`
yarn install
bundle exec passenger start
`

(last line instead of `passenger start`)

and then it worked!!!

**Comment by jywarren** [comment]
https://3000-bcf37db2-e88f-40ad-8f72-1239821526d6.ws-us02.gitpod.io/tags -- very nice!

**Comment by jywarren** [comment]
Added those last tweaks... hey tremendous job @keshavsethi -- thank you!

Fingers X this passes, then we'll be zooming along!

File: .gitpod.dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM gitpod/workspace-mysql
```
Comment (keshavsethi) [inline .gitpod.dockerfile:1]: @jywarren I think we can remove this file??

Comment (jywarren) [reply .gitpod.dockerfile:1]: possibly! want to try?

Comment (keshavsethi) [reply .gitpod.dockerfile:1]: @jywarren I checked gitpod after removing. Buil did getting failed so added it again.
Thanks!!

**Comment by jywarren** [comment]
Cool, so what ultimately happened? Looks like you added the dockerfile back in?

**Comment by jywarren** [comment]
OK, so, i had to run just plain `passenger start` and then I saw:

```
SQLite3::BusyException: database is locked: INSERT INTO "rsessions" ("session_id", "data", "created_at", "updated_at") VALUES (?, ?, ?, ?)
```

This can happen in sqlite but not in mysql, so i think we should try to use the default mysql account in the dockerfile-provided image. 

But, when i hit refresh, it worked: https://3000-ce5edf8f-1d90-42bf-b034-ddb10507ed95.ws-us02.gitpod.io/

This is just an intermittent issue with sqlite, so refreshing fixes it, but i think we'd better switch to mysql to avoid it altogether.

**Comment by jywarren** [comment]
OK, that boots up!

But I'm seeing this error, on many pages:

> Mysql2::Error: Expression # 28 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'plots.node.nid' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by: SELECT  *, rusers.status, MAX(node_revisions.timestamp) AS last_updated FROM `rusers` INNER JOIN `node` ON `node`.`uid` = `rusers`.`id` INNER JOIN `node_revisions` ON `node_revisions`.`nid` = `node`.`nid` WHERE (node_revisions.status = 1) AND (rusers.status = 1) GROUP BY rusers.id ORDER BY last_updated DESC LIMIT 30 OFFSET 0

**Comment by jywarren** [comment]
Wow, that leads us all the way back to: https://github.com/publiclab/plots2/issues/3120 - it relates to myqsl version number, and this apparently worked on version 5.7. https://github.com/publiclab/plots2/pull/4266#issuecomment-446462483

The version GitPod has is: `mysql  Ver 8.0.20-0ubuntu0.20.04.1 for Linux on x86_64 ((Ubuntu))`

Alternatively we could insist on `mariadb` which is what we use in production. 

Could we specify this via dockerfile? https://www.gitpod.io/blog/docker-in-gitpod/

i wonder -- could we use our existing Dockerfile? https://github.com/publiclab/plots2/blob/main/Dockerfile#L20 -- i guess?

**Comment by jywarren** [comment]
https://github.com/Piwigo/Piwigo/issues/376 suggests that the error we're seeing is due to strict mode. I'll try turning that off.

**Comment by jywarren** [comment]
Oh it's already off: `  strict: false`

**Comment by jywarren** [comment]
Added this line:

`mysql > SET @@global.sql_mode=(SELECT REPLACE(@@global.sql_mode, 'ONLY_FULL_GROUP_BY', ''));`

and it seems to work in the console! Let's see, fingers 🤞

**Comment by jywarren** [comment]
Hmm. i think it just timed out. I'll try to start it again...

**Comment by jywarren** [comment]
oh hmm, it started, and:

```

bash: syntax error near unexpected token `('
gitpod /workspace/plots2 $ 
gitpod /workspace/plots2 $ } && {
bash: syntax error near unexpected token `}'
gitpod /workspace/plots2 $ passenger start
bash: passenger: command not found
gitpod /workspace/plots2 $ 
gitpod /workspace/plots2 $ }
bash: syntax error near unexpected token `}'
gitpod /workspace/plots2 $ 
```
What's that from?

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+image:
+    file: .gitpod.dockerfile
+ports:
+    - port: 3000
+      onOpen: open-preview
+tasks:
+    - init: >
+        bundle install --without production &&
+        cp config/database.yml.gitpod config/database.yml &&
+        mysql > SET @@global.sql_mode=(SELECT REPLACE(@@global.sql_mode, 'ONLY_FULL_GROUP_BY', '')); &&
```
Comment (jywarren) [inline .gitpod.yml:10]: oh this is not formatted correctly...

**Comment by jywarren** [comment]
Hmm. synax still wrong:

```
mysql -e "SET @@global.sql_mode=(SELECT REPLACE(@@global.sql_mode, 'ONLY_FULL_GROUP_BY', ''));"
```

File: .gitpod.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+image:
+    file: .gitpod.dockerfile
+ports:
+    - port: 3000
+      onOpen: open-preview
+tasks:
+    - init: >
+        bundle install --without production &&
+        cp config/database.yml.gitpod config/database.yml &&
+        mysql -e "SET @@global.sql_mode=(SELECT REPLACE(@@global.sql_mode, 'ONLY_FULL_GROUP_BY', ''));"; &&
```
Comment (jywarren) [inline .gitpod.yml:10]: hmm there must be a syntax error in here, but I'm not sure where!

Comment (jywarren) [reply .gitpod.yml:10]: (Also depending on the mysql version, the syntax may be different, but I believe this one is good for us as I tested it on the console. Ref: https://stackoverflow.com/questions/23921117/disable-only-full-group-by)

Comment (jywarren) [reply .gitpod.yml:10]: @keshavsethi I kind of got stuck here but maybe you can figure out the next step? Or if @icarito has an idea?

File: .gitpod.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        mysql -e "CREATE DATABASE plots;"; &&
```
Comment (icarito) [inline .gitpod.yml:11]: I believe the issue is here, there's a `;` and then `&&`, this is a syntax error.

Comment (jywarren) [reply .gitpod.yml]: Hmm, after this fix we still see it:

```
gitpod /workspace/plots2 $  HISTFILE=/workspace/.gitpod/cmd-0 history -r; {
> bundle install --without production && cp config/database.yml.gitpod config/database.yml && mysql -e "SET @@global.sql_mode=(SELECT REPLACE(@@global.sql_mode, 'ONLY_FULL_GROUP_BY', ''));" && mysql -e "CREATE DATABASE plots;"; && cp db/schema.rb.example db/schema.rb && rake db:setup && yarn install
bash: syntax error near unexpected token `&&'
gitpod /workspace/plots2 $ 
gitpod /workspace/plots2 $ } && {
bash: syntax error near unexpected token `}'
gitpod /workspace/plots2 $ passenger start
bash: passenger: command not found
gitpod /workspace/plots2 $ 
gitpod /workspace/plots2 $ }
bash: syntax error near unexpected
```

But, actually there's a `;` on the next line too, so I'll try removing that.

**Comment by jywarren** [comment]
That did it. Perfect:

https://3000-a8cc69d8-8dd5-4128-93f5-1816a29b5f96.ws-us02.gitpod.io/

Screen Shot 2020-07-15 at 11 13 23 PM

**Comment by jywarren** [comment]
Thanks, everyone. This is really exciting. Adding a badge to the README too. 🎉 🎉 🎉 

@keshavsethi @Tlazypanda @cesswairimu @sagarpreet-chadha @Shulammite-Aso @keshav234156 @NitinBhasneria @shreyaa-sharmaa @SidharthBansal if you rebase or resync your PRs on `plots2` now, you should get a GitPod build here too! 

Great for testing integrations, checking UI changes on PRs (esp from first-timers), all kinds of things. Woohoo!!!!

**Comment by jywarren** [comment]
Aww, it looks like we didn't actually completely solve https://github.com/publiclab/plots2/pull/8145#issuecomment-658443873 -- I'm trying to work on it a bit more in #8152 now!

**Comment by sagarpreet-chadha** [comment]
Wow this is so cool!!! 
So we have a docker container running in server, which is listening to github PR's using github webhooks, right?

**Comment by jywarren** [comment]
I believe so but basically it's hosted at GitPod, free!

**Comment by jywarren** [comment]
Just confirming that we see an error on https://3000-ff07100e-6fdf-4fbe-b308-983437a28749.ws-us02.gitpod.io/wiki -- 

```
Mysql2::Error: Expression #26 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'plots.node_revisions.vid' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by: SELECT  `node`.`nid` AS t0_r0, `node`.`vid` AS t0_r1, `node`.`type` AS t0_r2, `node`.`language` AS t0_r3, `node`.`title` AS t0_r4, `node`.`uid` AS t0_r5, `node`.`status` AS t0_r6, `node`.`created` AS t0_r7, `node`.`changed` AS t0_r8, `node`.`comment` AS t0_r9, `node`.`promote` AS t0_r10, `node`.`moderate` AS t0_r11, `node`.`sticky` AS t0_r12, `node`.`tnid` AS t0_r13, `node`.`translate` AS t0_r14, `node`.`cached_likes` AS t0_r15, `node`.`comments_count` AS t0_r16, `node`.`drupal_node_revisions_count` AS t0_r17, `node`.`path` AS t0_r18, `node`.`main_image_id` AS t0_r19, `node`.`slug` AS t0_r20, `node`.`views` AS t0_r21, `node`.`latitude` AS t0_r22, `node`.`longitude` AS t0_r23, `node`.`precision` AS t0_r24, `node_revisions`.`vid` AS t1_r0, `node_revisions`.`nid` AS t1_r1, `node_revisions`.`uid` AS t1_r2, `node_revisions`.`title` AS t1_r3, `node_revisions`.`body` AS t1_r4, `node_revisions`.`teaser` AS t1_r5, `node_revisions`.`log` AS t1_r6, `node_revisions`.`timestamp` AS t1_r7, `node_revisions`.`format` AS t1_r8, `node_revisions`.`status` AS t1_r9 FROM `node` LEFT OUTER JOIN `node_revisions` ON `node_revisions`.`nid` = `node`.`nid` WHERE (node_revisions.status = 1 AND node.status = 1 AND (type = 'page' OR type = 'tool' OR type = 'place')) GROUP BY node_revisions.nid ORDER BY node_revisions.timestamp DESC LIMIT 30 OFFSET 0

```
line 17 of Trace of template inclusion: app/views/wiki/index.html.erb
```

15 16 17 18 19 20 | <th><a href = "?sort=likes"> <%= translation('wiki._wikis.likes') %></a> <i class="fa fa-sort" aria-hidden="true"></i></th>     </tr>     <% wikis.each do \|wiki\| %>       <tr>         <td style="width:50%"><i class="fa fa-file"></i> <a href="<%= wiki.path %>"><%= wiki.latest.title %></a></td>         <td><i><%= wiki.path %></i></td>
-- | --


```

**Comment by icarito** [comment]
yes, some of our queries seem to be incompatible with MySQL 5.7 - I ran into this with our recent upgrade. In Cloud SQL I added the flag: sql_mode="TRADITIONAL" and it started working.

**Comment by jywarren** [comment]
MySQL 5.7 and higher? Thanks for the tip, i'll try that!

**Comment by jywarren** [comment]
addressing in https://github.com/publiclab/plots2/pull/8152

**Pull request merged**